File: java/6.0/jaxb/test1/Item.java

import javax.xml.bind.annotation.*;

@XmlType
public class Item {

	private String name;

	public Item() {
	}

	public Item(String name) {
		this.name = name;
	}

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}
}

File: go/immutable/struct_immutable/item/item.go
package item

type Item interface {
	Name() string
	Value() int32
}

type item struct {
	name string
	value int32
}

func (i item) Name() string {
	return i.name
}

func (i item) Value() int32 {
	return i.value
}

func NewItem(name string, value int32) Item {
	return item{name, value}
}
